libs/fetcher.client.js

```diff
@@ -152,7 +152,9 @@ Request.prototype.end = function (callback) {
             debug('Executing request %s.%s with params %o and body %o', self.resource, self.operation, self._params, self._body);
             setImmediate(executeRequest, self, resolve, reject);
         });
-        promise = promise.then(captureMeta, captureMeta);
+        promise = promise.then(captureMeta, function (err) {
+            return Promise.reject(captureMeta(err));
```

Comment:
In our server implementation (https://github.com/yahoo/fetchr/blob/master/libs/fetcher.js#L153) we just re-throw the error so I think to be consistent you can just do`throw captureMeta(err);` here.